app/main.py

```diff
@@ -1 +1,48 @@
-#  write your code here
+
+class Animal:
+    def __init__(
+            self,
+            name: str,
+            appetite: int,
+            is_hungry: bool = True
+    ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        else:
+            return 0
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=3)
```

Comment:
This logic is not right, in case you take `is_hungry` argument, you should pass it to the `super().__init__(` as well. There is no need to redefine arguments on lines 28 and 29/ Make similar changes in the `Dog` class as well

Reply:
> This logic is not right, in case you take `is_hungry` argument, you should pass it to the `super().__init__(` as well. 

I should pass all argument, that take my __init__ in super().__init__?

Reply:
>There is no need to redefine arguments on lines 28 and 29/ Make similar changes in the `Dog` class as well

`    def __init__(self, name: str, is_hungry: bool = True) -> None:
        super().__init__(name, appetite=3, is_hungry=True)
        self.name = name
        self.is_hungry = is_hungry

    @staticmethod
    def catch_mouse() -> None:
        print("The hunt began!")


class Dog(Animal):
    def __init__(self, name: str, is_hungry: bool = True) -> None:
        super().__init__(name, appetite=7, is_hungry=True)
        self.name = name
        self.is_hungry = is_hungry`
  
if I remove `self.is_hungry = is_hungry` from Dog - all right!
but i I remove `self.is_hungry = is_hungry` from Cat - error (    def test_feed_animals_function(animals, total_food_points):
>       assert main.feed_animals(animals) == total_food_points
E       assert 3 == 0
E         +3
E         -0
)